### PR TITLE
[Helm] Error using default controllerManagerConfigYaml

### LINKED ISSUE #1315: helm install Error using default controllerManagerConfigYaml
**What happened**:
When using Helm to install kueue, Error happens:
`# helm install kueue kueue/ --namespace kueue-system `
```
Error: INSTALLATION FAILED: template: kueue/templates/webhook/webhook.yaml:120:84:
executing "kueue/templates/webhook/webhook.yaml" at <$integrationsConfig.podOptions>:
wrong type for value; expected map[string]interface {}; got interface {}
```
in [webhook.yaml#L120:](https://github.com/kubernetes-sigs/kueue/blob/main/charts/kueue/templates/webhook/webhook.yaml#L120)

`{{- if and (hasKey $integrationsConfig "podOptions") (hasKey $integrationsConfig.podOptions "namespaceSelector") }}`

When [integrations.podOptions](https://github.com/kubernetes-sigs/kueue/blob/main/charts/kueue/values.yaml#L97) is not set(by default), the expression above caused this Error below: 

`$integrationsConfig.podOptions got interface {}`

**What you expected to happen**:
This Error shoud not happen when default config is used.

**How to reproduce it (as minimally and precisely as possible)**:
```
$ git clone https://github.com/kubernetes-sigs/kueue.git
$ cd kueue/charts
$ helm install kueue kueue/ --create-namespace --namespace kueue-system
```

**Anything else we need to know?**:
related PR:
#1247

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`): main
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by stuton**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Parentheses around the optional dict as a mean of null-safety

#### Which issue(s) this PR fixes:

Fixes #1315

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: charts/kueue/templates/webhook/webhook.yaml [hunk 1]
```diff
@@ -321,7 +321,7 @@ webhooks:
   {{- end }}
   name: vpod.kb.io
   namespaceSelector:
-    {{- if and (hasKey $integrationsConfig "podOptions") (hasKey $integrationsConfig.podOptions "namespaceSelector") }}
+    {{- if and (hasKey $integrationsConfig "podOptions") (hasKey ($integrationsConfig.podOptions) "namespaceSelector") }}
```
Comment (kerthcet) [inline charts/kueue/templates/webhook/webhook.yaml:324]: why `$integrationsConfig` doesn't have the parenthesis as well?

Comment (kerthcet) [reply charts/kueue/templates/webhook/webhook.yaml:324]: Got the point, forget it.

**Review by mimowo — COMMENTED** [review]
/lgtm
/assign @tenzen-y

**Review by tenzen-y — COMMENTED** [review]
Thanks!
/approve

**Pull request merged**